crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -227,6 +227,46 @@ impl Format<PyFormatContext<'_>> for FormatDanglingComments<'_> {
     }
 }
 
+/// Formats the dangling comments within a parenthetical, for example:
+/// ```python
+/// [  # comment
+///     1,
+///     2,
+///     3,
+/// ]
+/// ```
+pub(crate) fn parenthetical_comments(comments: &[SourceComment]) -> FormatParentheticalComments {
+    FormatParentheticalComments { comments }
+}
+
+pub(crate) struct FormatParentheticalComments<'a> {
+    comments: &'a [SourceComment],
+}
+
+impl Format<PyFormatContext<'_>> for FormatParentheticalComments<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext>) -> FormatResult<()> {
+        let mut first = true;
+
+        for comment in self
```

Comment:
It would make sense to add a debug assertion that `comment` is a `end_of_line` comment because this implementation doesn't handle own line comments correctly.